storage/memoized_iterator.go

```diff
@@ -20,7 +20,9 @@ import (
 	"github.com/prometheus/prometheus/tsdb/chunkenc"
 )
 
-// MemoizedSeriesIterator wraps an iterator with a buffer to look back the previous element.
+// MemoizedSeriesIterator wraps an iterator with a buffer to look back the previous element. This iterator regards
+// integer histograms as float histograms. Calls to Seek() will never return chunkenc.Histogram and it expects
+// that the chunkenc.Iterator provided to it can perform the conversion directly with a call to AtFloatHistogram().
```

Comment:
Which is the contract anyway, see https://github.com/prometheus/prometheus/blob/0ab95536115adfe50af249d36d73674be694ca3f/tsdb/chunkenc/chunk.go#L130

Reply:
Maybe note here that `MemoizedSeriesIterator` deliberately does not implement `chunkenc.Iterator`.